### DISCUSSION TITLE
How to make invoked machines show up in the Visualizer's Actors tab

### DISCUSSION THREAD

**Discussion by jericdeleon** [statelyai/xstate#3451] [Q&A]
👋 Hello!

I have a sample sandbox where the current behavior is:
- A top-level machine is setup for visualization through `devTools: true`

This top-level machine has two steps:
  - On the first step, it invokes a separate machine
  - On the second step, it spawns another separate machine

On the visualizer, as we go through the steps: 
- Only the top-level machine and the spawned machine show up on the Actors tab
- The invoked machine cannot be visualized

Is there a way to visualize the invoked machine and make it show up on the Actors tab?
The logic I'm trying to model really makes more sense with an invoked machine, and I'd like to avoid transforming it into a spawned machine just for visualization purposes (totally worth it, just wish the tradeoff isn't necessary).

Thanks for the help in advance!

Link to the sandbox: https://codesandbox.io/s/eloquent-framework-8yu4q9
If the sandbox is not working, here's the source repo: https://github.com/jericdeleon/xstate-inspect-vue

**Comment by davidkpiano** [comment]
That's strange that it doesn't visualize the invoked machine. We are planning to rework `@xstate/inspect` which will make it easier to inspect/debug what is going on.

One way to debug it right now is to add an event listener for `window.addEventListener('message', ...)` and see if the invoked machine gets sent over.

**Reply by jericdeleon** [reply]
Thanks for the debugging tip!

Excited for the reworked release, hopefully it'll show up by then 🤞
